### PR TITLE
feat: guided tour component

### PR DESCRIPTION
**Pull request by rohanchkrabrty**
## Summary

- Add the guided **Tour** component (RFC 003) for building step-by-step onboarding flows, built on Base UI Popover with a spotlight overlay
- Split the implementation into composable parts (`tour-root`, `tour-content`, `tour-overlay`, `tour-parts`, `tour-context`) with supporting hooks (`use-tour-target`, `use-prefers-reduced-motion`) and shared types
- Export the component from the package root and add full test coverage for stepping, targeting, and overlay edge cases
- Wire up docs and playground: MDX guide, props reference, live demo, and interactive tour examples on the www site

### CODE REVIEW

File: packages/raystack/components/tour/tour-overlay.tsx [hunk 1]
```diff
@@ -0,0 +1,280 @@
...
+  y: number;
+  width: number;
+  height: number;
+}
+
+/**
+ * Whether two rects match within half a pixel on every side. Shared with the
+ * root's reveal loop, which feeds it `DOMRect`s (`x`/`y` mirror `left`/`top`).
+ */
+export function rectsEqual(a: SpotlightRect, b: SpotlightRect) {
```
Comment (Shreyag02) [inline packages/raystack/components/tour/tour-overlay.tsx]: Move rectsEqual (and SpotlightRect) out of the overlay into a shared utils.ts.
Right now rectsEqual lives in the overlay component but tour-root.tsx imports it — so the root is reaching into a leaf component to grab geometry helpers. 

That's backwards: the parent shouldn't depend on internals of a child.

File: packages/raystack/components/tour/tour-root.tsx [hunk 2]
```diff
@@ -0,0 +1,476 @@
...
+    if (!open) {
+      lastActiveIndexRef.current = -1;
+      return;
+    }
+    if (!popoverOpen || !step || lastActiveIndexRef.current === index) return;
+    lastActiveIndexRef.current = index;
+    emit({ type: 'step:active', index, step });
+  }, [open, popoverOpen, index, step, emit]);
+
+  // Scroll the target into view once, when the step becomes active.
```
Comment (Shreyag02) [inline packages/raystack/components/tour/tour-root.tsx]: Since step is an object rebuilt on every render (all our examples pass steps inline), this effect fires on every host re-render — and if the target is out of view, it calls scrollIntoView again, moving the page back while the user is scrolling mid-step.

should swap step for index in the deps (keep anchor so late-resolving targets still scroll in).

File: packages/raystack/components/tour/use-tour-target.ts [hunk 3]
```diff
@@ -0,0 +1,134 @@
...
+      return;
+    }
+    if (target == null) {
+      // Detached step: nothing to resolve, the popover renders centered.
+      setElement(null);
+      setState('found');
+      return;
+    }
+
+    let timer: ReturnType<typeof setTimeout> | undefined;
```
Comment (Shreyag02) [inline packages/raystack/components/tour/use-tour-target.ts:57]: The timer slot is cleared before the callback runs, so if unrelated DOM churn triggers another reconcile() while the target is still missing, a fresh timeout gets scheduled and onNotFound fires again — meaning a possible double skip/stop.

File: packages/raystack/components/tour/use-tour-target.ts [hunk 4]
```diff
@@ -0,0 +1,134 @@
...
+    } catch {
+      // A throwing resolver (e.g. `() => editor.getNode('x')` before the editor
+      // is ready) should degrade to "not found", not crash the tour — matching
+      // how an invalid selector is handled above.
+      return null;
+    }
+  }
+  if (target instanceof Element) return target;
+  return target.current ?? null;
+}
```
[the remark this answers is not in the document]
Comment (Shreyag02) [reply packages/raystack/components/tour/use-tour-target.ts]: should add the suggested fix

**Review by Shreyag02 — CHANGES_REQUESTED** [review]

File: packages/raystack/components/tour/use-tour-target.ts (hunk 3 above)
Comment (rohanchkrabrty) [reply packages/raystack/components/tour/use-tour-target.ts:57]: This shouldn't fire twice in practice. When the timeout triggers, `onNotFound` advances the tour or closes it, which changes the effect's dependencies and immediately cleans up the observer and timer.

Comment (Shreyag02) [reply packages/raystack/components/tour/use-tour-target.ts:57]: got it

**Review by Shreyag02 — APPROVED** [review]
